src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -102,6 +103,22 @@ public void deregisterCryptoKeyProvider(CryptoKeySpi keyProvider) {
         return provider.getKeyManagers(privateKeyUri, certificateUri);
     }
 
+    /**
+     * Get JCA KeyManagers, used for Secret manager.
+     *
+     * @param privateKeyUri private key URI
+     * @return KeyManagers that manage the specified private key
+     * @throws ServiceUnavailableException if crypto key provider service is unavailable
+     * @throws KeyLoadingException if crypto key provider service fails to load key
+     * @throws URISyntaxException if key URI syntax error
+     */
+    public KeyPair getKeyPair(String privateKeyUri)
```

Comment:
I am confused again. :) How would secret manager provide the privateKeyUri?

Reply:
Like we discussed offline, SecurityService will add a method to get the device identity URI